common/core/desktop/tests/unit/kmnkbd/test_kmx_xstring.cpp

```diff
@@ -16,7 +16,587 @@
 using namespace km::kbp::kmx;
 using namespace std;
 
-void test_incxstr() {
+PKMX_WCHAR find_ptr_to_last_character(PKMX_WCHAR p_first) {
+  int length = wcslen((const wchar_t *)p_first); 
+ if (length > 0)
+    return p_first + length - 1;
+  else
+    return p_first;
+}
+
+
+void test_decxstr() {
+    
+    PKMX_WCHAR p_start; // pointer start of input 
+    PKMX_WCHAR p;       // pointer end of input
+    PKMX_WCHAR q;       // pointer output
+
+
+    // -------------------------------------------------------------------------------------------------------------------------------------------------------
+    // more tests: check if we might end up left of pstart
+    // -------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    p_start = (PKMX_WCHAR)u"\u0001";
+    p       = find_ptr_to_last_character(p_start);
```

Comment:
Since this is a unit test where we know the content of the pointer it might be better (aka easier to read) to explicitly state where `p` points to, i.e. in this case `p = p_start`. On the other hand this looses the reminder that `p` points to the last character.
Another option would be to assert `p`:
```
p       = find_ptr_to_last_character(p_start);
assert(p == p_start);
```
but I don't particularly like that option either...

Reply:
I am happy with the consistency of the test setup as it is for this.

Reply:
...did no changes...